### ISSUE TITLE
Training diffusion model resulting noise

### ISSUE THREAD

**Issue by kenkalang** [yukara-ikemiya/friendly-stable-audio-tools#4]
I've been trying to train diffusion model with stable audio 1.0 config, I also trained the autoencoder with the Stable Audio 1.0 VAE for 50k steps [autoencoder result](https://storage.googleapis.com/wandb-production.appspot.com/kenkalang/stable_model/n43uce44/media/audio/recon_50025_5f76a8daf5fea2778e82.wav?Expires=1723351448&GoogleAccessId=gorilla-files-url-signer-man%40wandb-production.iam.gserviceaccount.com&Signature=Y%2FMkZvVMKO3LRbAGLDD9SKa6D2q5gE%2F%2Fk42KZQj7fjH7o8dHaW44oI8AjoE5ss2gujVXUrE1yml%2BDaLZlc2Nm7Li18d%2Bl1l5j5Birh0%2BZ9po5lgPrvcW0hVjPESvNg%2FUly7hqMDiObL1ohOZyUqHrAHpNRdhNellH2vdwLPkTnJ8Ds8xAMJstjx6M1TLSPLKWp7sf9LcmPV%2FC9QUkr4gubohrV2h7RqmcwKkpZSBrnX%2BzB4wsrdIsKlSneml6nk7MsxicobIQzr7Qm0CeKVuYSLeGAl7YbpHjVwv2H%2FJXVai%2FeeM0Mlz6VIV70FFrfSNhNVQEllwIHJ%2B8uuGt6maZQ%3D%3D) which I then used as the pre-transform ckpt. However, after training the diffusion model for 100k steps, it still results in noise [diffusion result](https://storage.googleapis.com/wandb-production.appspot.com/kenkalang/stable_diffusion/6uljwfpe/media/audio/demo_cfg_1.0_30015_51322e46a3ec63628c5f.wav?Expires=1723351230&GoogleAccessId=gorilla-files-url-signer-man%40wandb-production.iam.gserviceaccount.com&Signature=mBuOedab26FbPg8yNLnvEoqenmX%2FeYCTZq2VP6CsZEmnZ4yo8IYkzMPhoFTVL1pY7ZB9kwdpb%2FWv%2BXToKmHY54ljD%2B26rfj8upy8oG6a%2BwiDuOs%2FqiDG0jqmpnP%2FGsY05AAap2DKfIOnB3qjLRL4Vhn%2F2qL8QPeewevM5A6fw%2BjhOv%2Fd8wJcgS7cOKDGMLOT%2Ff0JIbY8DGuKLzCF2gjhwcsZzDiytRiuzB2RqSKuF9Xlf26hO5EY8arqF75li7XPHFCfjVcIDoTehpAm7byNnN3EwuurjCwY86bWp8mVp%2F4jYWDpBFljgMRFoentz3Y7Id8aQ0RBm50gTCNz5bnZUw%3D%3D). The dataset contains 2,500 hours of songs. I used the same CLAP model as you recomended. i used deepspeed as my training strategy. Is there anything I might have missed in my training?

**Comment by yukara-sony** [comment]
Hi, thank you for using my repository.
I haven't tried Stable Audio 1.0 training, but I can help you find the reason.
I believe while training the job should save logs at `WandB`. Could you share some logs of your training such as reconstruction samples, loss curve, and input std? 

Btw, some possible issues would be:
1. VAE checkpoint is not loaded correctly.
2. Text prompts are not properly fed into a model while training.

**Comment by yukara-ikemiya** [comment]
Hi, thank you for using my repository.
I haven't tried Stable Audio 1.0 training, but I can help you find the reason.
I believe while training the job should save logs at WandB. Could you share some logs of your training such as reconstruction samples, loss curve, and data std?

Btw, some possible issues would be:
1. VAE checkpoint is not loaded correctly.
2. Text prompts are not properly fed into a model while training.

**Comment by kenkalang** [comment]
Thanks for your response,

![image](https://github.com/user-attachments/assets/59eb7e1f-8514-42c5-8cbc-f0489d3a847f)

https://github.com/user-attachments/assets/38b2a1d9-ab18-43cf-9938-bc79e4fd8494

I noticed some irregularities in the training logs and was optimistic about achieving a good output by 100k steps. I followed the steps outlined in the repository, and everything worked fine with the text prompts, as I confirmed by logging them. I indeed suspect that the VAE might not be loading correctly. How can I verify if the VAE is being loaded properly?

**Comment by yukara-ikemiya** [comment]
Assuming that the VAEs of Stable Audio 1.0 and 2.0 are slightly different, when I tested the training of the SA 2.0 VAE, the `data_std` tended to be around 0.9. Therefore, the logged `data_std` value might be suspicious..?

In my training code, I believe reconstruction samples are save to WandB under the name `Reconstruction (Pretransform)` as well as the above logs. Can you find them in the WandB log?

**Comment by kenkalang** [comment]
I deleted all my previous files since they weren't working anymore, so I retrain the VAE 2.0 following your README instructions and here is the recon pretransform 

[original]
https://github.com/user-attachments/assets/b5264eee-8d18-4d41-b87d-7f0c7195b51b

[recon]

https://github.com/user-attachments/assets/7a5eac79-f8cd-4ced-bafc-4912a3a9d219

 . My question is, could using the DeepSpeed strategy affect the results? Someone mentioned that when they used DeepSpeed, the inference results were noisy, although the demo sounded fine. However, when they used DDP, it worked perfectly. Lastly, may I know the configuration such as batch size, etc and device that you used?

**Comment by yukara-ikemiya** [comment]
Sorry for the late reply.
I believe DeepSpeed should not affect training results dractically.
If DeepSpeed affected a lot, there might be a bug in checkpoint saving/loading..?

Unfortunately, since I have never tried DeepSpeed options of Lightning, I have no idea if a DeepSpeed implementation in stable-audio-tools has a problem.

Regarding the reconstruction sample you pasted, from my experience of VAE training, the quality should be able to be improved if the model is trained for longer time.

**Comment by twobob** [comment]
I did train VAE 1_0 for 110k steps.  
it was still pretty sizzly.  ;)   
It is what it is - in my (albeit limited) experience it was things with vocals that are more mashed up

conversion
https://github.com/user-attachments/assets/9cc7dfcf-8193-4791-8fe9-c140a807d3db

original
https://github.com/user-attachments/assets/af0040c1-e94c-40da-a29b-2bddca625dd2

:) This wasn't a complaint, simply further information.

**Comment by yukara-ikemiya** [comment]
I think the reconstruction quality strongly depends on dataset size and training time.
It is common that audio datasets don't contain vocal tracks and the quality of vocals would be especially degraded.

In the Stable Audio 2.0 paper, it is written that they trained the VAE model for 16000 GPU hours...
https://arxiv.org/abs/2404.10301
Based on my experience, I also feel VAE requires training for at least 1 million steps to reach a certain level of performance.

So I guess you may have to keep training the model.
And I believe that by checking whether the losses (e.g. mrstft_loss, loss_adv) has converged, we can determine whether the training has progressed to a certain extent or is still in its early stages.

**Comment by twobob** [comment]
yeah. I helped out on their discord as a harmonai guide the year before fwiw. They have a lot of iron.  
I was indeed [eyeballing the loss](https://wandb.ai/twobob/vae_training/runs/6wpz36ya?nw=nwusertwobob) and it was AFAIK still moving...  It was only one night of training (on   [DATASET](https://www.kaggle.com/datasets/twobob/moar-bobtex-n-friends-gpu-fodder):  30Gb expanded to 250Gb of samples processed with [aeiou](https://github.com/drscotthawley/aeiou)) to get 110k on version 1_0. 
just throwing out there that 110k steps probably isnt enough if choirs are "your thing"  but might be perfectly decent if you like WubWubs and Squawks . 

I'm thinking I might just go with training a 2_0 for a week or so.

Maybe I should open another issue so OP doesnt get pinged.  but it seemed relevant
